python_modules/dagit/dagit/pipeline_run_storage.py

```diff
@@ -50,6 +54,35 @@ def __init__(self, run_id, pipeline_name, config, execution_plan):
         self.pipeline_name = pipeline_name
         self.config = config
         self.execution_plan = execution_plan
+        self._log_queue_lock = gevent.lock.Semaphore()
+        self._log_queue = []
+        self._flush_queued = False
+        self._queue_timeout = None
+
+    def _enqueue_flush_logs(self):
+        with self._log_queue_lock:
+            if not self._flush_queued:
+                self._queue_timeout = time.time()
```

Comment:
we could move this outside the lock scope (the time.time() call)

Reply:
I think it's fine there, just so we couldn't override it by accident if logic changes at some point.